src/dirs.proj

```diff
@@ -19,11 +19,44 @@
     <Project Include="XMakeBuildEngine\UnitTestsPublicOM\Microsoft.Build.Engine.OM.UnitTests.csproj" />
   </ItemGroup>
 
+  <Import Project="dir.targets" />
   <Import Project="..\dir.traversal.targets" />
 
   <PropertyGroup>
+    <SkipVersionGeneration>true</SkipVersionGeneration>
+    <!-- Turn off the automated package generation in packages.targets:AddNuGetPackageVersionMetadataToNuspecs;
+         we'll use the one produced by NerdBank.GitVersioning instead. -->
+    <DoNotGeneratePackageVersion>true</DoNotGeneratePackageVersion>
+  </PropertyGroup>
+
+  <Import Project="$(ToolsDir)packages.targets" Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'" />
+
+  <PropertyGroup Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'">
+    <TraversalBuildDependsOn>
+      $(TraversalBuildDependsOn);
+      GetNuspecProperties;
+      BuildPackages;
+    </TraversalBuildDependsOn>
+    <!-- We use Microsoft.DotNet.Build tools to create our packages, but we want the version comes from 
+         NerdBack.GitVersioning. -->
+    <GetNuGetPackageVersionsDependsOn>SetPackageVersion</GetNuGetPackageVersionsDependsOn>
+  </PropertyGroup>
+
+  <PropertyGroup>
     <!-- Explicitly set the OutDir as it is used by the packaging targets -->
     <OutDir>$(BaseOutputPathWithConfig)</OutDir>
   </PropertyGroup>
 
+  <Target Name="SetPackageVersion" DependsOnTargets="GetNuGetPackageVersion" Returns="$(PackageVersion)">
```

Comment:
Do you need the `Returns`? That should be the case only if you call this from another project.